sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetRelation.scala

```diff
@@ -306,6 +315,10 @@ private[sql] class DefaultSource
       SQLConf.PARQUET_INT96_AS_TIMESTAMP.key,
       sqlContext.conf.getConf(SQLConf.PARQUET_INT96_AS_TIMESTAMP))
 
+    // Whole stage codegen (PhysicalRDD) is able to deal with batches directly
+    parquetConf.setBoolean(ParquetRelation.RETURNING_BATCH,
+      supportBatch(sqlContext, StructType(partitionSchema.fields ++ dataSchema.fields)))
```

Comment:
In general, I'd prefer if all new code passes options explicitly to the closure instead of through the conf.